boards/arm/nucleo_h743zi/nucleo_h743zi.dts

```diff
@@ -81,7 +81,8 @@
 	};
 };
 
-&adc1_2 {
+&adc1 {
+	pinctrl-0 = <&adc1_inp15_pa3>;
 	status = "okay";
```

Comment:
I'm ok if you want to change the ADC definition other here, but please also update the index.rst for this board. I have this board so I'm going to try your PR when I have time today.

Reply:
I actually think it was never the intention to use the special dual-mode ADC here. And I guess our driver doesn't support that properly either. That's why I changed to &adc1.

Updated the index.rst aswell. Would be nice if you could verify the PR, as I don't have this board. Thanks!

Reply:
Agreed, I will approve once I have tested with my board

Reply:
@gmarull Are you fie with this since you introduced this node ?

Reply:
ADC on H7 used to work fine (using it on a custom board). I'll test it on my nucleo with and without these changes.

Reply:
Ok, I have added a DNM. Can you please remove it once you're done with your test?

Reply:
Ok, so it turns out `https://github.com/zephyrproject-rtos/zephyr/blob/master/tests/drivers/adc/adc_api/src/test_adc.c#L201` should be `15` and not `0`. Once this is changed test seems to report coherent values. Should we fix it here or on a new PR? It's not related to the current changes.

Reply:
Thanks for testing. I suggest to get this one merged first and have a dedicated PR for that issue. I actually think that the ADC test should be completely rewritten so that it obtains the used ADC channel from DTS instead of hard-coding it.

So I will remove the DNM again.

Reply:
> Ok, so it turns out `https://github.com/zephyrproject-rtos/zephyr/blob/master/tests/drivers/adc/adc_api/src/test_adc.c#L201` should be `15` and not `0`. Once this is changed test seems to report coherent values. Should we fix it here or on a new PR? It's not related to the current changes.

Good catch, actually that was the problem ! 
I changed to 15 in test/adc/ac_api and using this PR:
```c
#elif defined(CONFIG_BOARD_NUCLEO_H743ZI)
#define ADC_DEVICE_NAME         DT_LABEL(DT_INST(0, st_stm32_adc))
#define ADC_RESOLUTION		16
#define ADC_GAIN		ADC_GAIN_1
#define ADC_REFERENCE		ADC_REF_INTERNAL
#define ADC_ACQUISITION_TIME	ADC_ACQ_TIME_DEFAULT
#define ADC_1ST_CHANNEL_ID	15
```
Result with tying PA3 (A0) to VCC:
```
*** Booting Zephyr OS build zephyr-v2.4.0-550-g8ae325534e06  ***
Running test suite adc_basic_test
===================================================================
START - test_adc_sample_one_channel
Samples read: 0xffffffff 0xffff8000 0xffff8000 0xffff8000 0xffff8000 0xffff8000
 PASS - test_adc_sample_one_channel
===================================================================
START - test_adc_sample_two_channels
 SKIP - test_adc_sample_two_channels
===================================================================
START - test_adc_asynchronous_call
Samples read: 0xffffffff 0xffffffff 0xffffffff 0xffffffff 0xfffffffe 0xffff8000
 PASS - test_adc_asynchronous_call
...
```
This reads 0xFF ish as expected.

Tying to GND now:
```
*** Booting Zephyr OS build zephyr-v2.4.0-550-g8ae325534e06  ***
Running test suite adc_basic_test
===================================================================
START - test_adc_sample_one_channel
Samples read: 0x0000 0xffff8000 0xffff8000 0xffff8000 0xffff8000 0xffff8000
 PASS - test_adc_sample_one_channel
===================================================================
START - test_adc_sample_two_channels
 SKIP - test_adc_sample_two_channels
===================================================================
START - test_adc_asynchronous_call
Samples read: 0x0000 0x0000 0x0000 0x0000 0x0002 0xffff8000
 PASS - test_adc_asynchronous_call
===================================================================
...
```
This read 0x00 ish as expected ! 😄

Reply:
> Thanks for testing. I suggest to get this one merged first and have a dedicated PR for that issue. I actually think that the ADC test should be completely rewritten so that it obtains the used ADC channel from DTS instead of hard-coding it.


So now I fully approve this PR, all is working fine when the test actually uses the correct channel 🙃 . Agreed on fixing the ADC test in another PR.